rust/socket-factory/src/lib.rs

```diff
@@ -50,6 +54,9 @@ pub fn udp(std_addr: &SocketAddr) -> io::Result<UdpSocket> {
     socket.set_nonblocking(true)?;
     socket.bind(&addr)?;
 
+    socket.set_send_buffer_size(SEND_BUFFER_SIZE)?;
+    socket.set_recv_buffer_size(RECV_BUFFER_SIZE)?;
+
     let send_buf_size = socket.send_buffer_size()?;
     let recv_buf_size = socket.recv_buffer_size()?;
 
```

Comment:
Any value here you think in reporting this to Sentry so we have a rough idea of the actual size the OS has given us per platform?

Reply:
We can change the log-level to DEBUG but it is quite spammy because we create a new UDP socket for each forwarded DNS query (to ensure we use randomized ports). It should be the same value for all sockets really so perhaps I can log them every time we rebind the p2p sockets.